tipg/middleware.py

```diff
@@ -41,3 +46,24 @@ async def dispatch(self, request: Request, call_next):
                 response.headers["Cache-Control"] = self.cachecontrol
 
         return response
+
+
+class CatalogUpdateMiddleware(BaseHTTPMiddleware):
+    """Middleware to update the catalog cache."""
+
+    async def dispatch(self, request: Request, call_next):
+        """Fetch Catalog either immediately or in background."""
+        if request.query_params.get("refresh"):
+            logger.debug("Forcing catalog refresh.")
+            await register_collection_catalog(request.app)
+            response = await call_next(request)
+        elif datetime.now() > request.app.state.catalog_expire:
+            logger.debug("Running catalog refresh in background.")
+            response = await call_next(request)
+            response.background = BackgroundTask(
```

Comment:
One of my favorite things in programming is when folks choose words that obscure how it actually works 😆 . A `BackgroundTask` is just a callback [AFAICS](https://github.com/encode/starlette/blob/4bab981d9e870f6cee1bd4cd59b87ddaf355b2dc/starlette/responses.py#L173-L174).